### PR TITLE
JBPM-5587 - Stunner - add OnEntry/OnExit action properties to Busines…

### PR DESCRIPTION
**Pull request by tsurdilo**
…s Rule Task, Script Task, and User Task

### CODE REVIEW

**Comment by tsurdilo** [comment]
jenkins please retest this

**Comment by tsurdilo** [comment]
@romartin @pefernan Hi guys, I think @jlindop is out for a while and I have more prs that depend on this one. Would one of you please review this for me? I have tested it in workbench on wildfly10 already and it works fine so maybe reviewing code is enough in case you don't have time for test inside AS. Thanks!

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/BusinessRuleTaskExecutionSet.java [hunk 1]
```diff
@@ -55,22 +57,75 @@
...
-            afterElement = "ruleFlowGroup"
+            afterElement = "scriptLanguage"
     )
     @Valid
     private IsAsync isAsync;
 
+    @Property
+    @FormField(
+            type = CheckBoxFieldType.class,
+            labelKey = "adHocAutostart",
```
Comment (pefernan) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/BusinessRuleTaskExecutionSet.java]: Same here, this lable key must be added to the 18n Bundle

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/BusinessRuleTaskExecutionSet.java [hunk 2]
```diff
@@ -55,22 +57,75 @@
 
     @Property
     @FormField(
+            type = TextAreaFieldType.class,
+            labelKey = "onentryaction",
```
Comment (pefernan) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/BusinessRuleTaskExecutionSet.java]: I think that this key isn't present on the I18n bundle, please add it here: https://github.com/tsurdilo/kie-wb-common/blob/JBPM-5587-master/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/main/resources/org/kie/workbench/common/stunner/bpmn/client/resources/i18n/StunnerBPMNConstants.properties

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/BusinessRuleTaskExecutionSet.java [hunk 3]
```diff
@@ -55,22 +57,75 @@
...
+            afterElement = "ruleFlowGroup",
+            settings = {@FieldParam(name = "rows", value = "5")}
+    )
+    @Valid
+    private OnEntryAction onEntryAction;
+
+    @Property
+    @FormField(
+            type = TextAreaFieldType.class,
+            labelKey = "onexitaction",
```
Comment (pefernan) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/BusinessRuleTaskExecutionSet.java]: Same here, this lable key must be added to the 18n Bundle

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/OnEntryAction.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+@Portable
+@Bindable
+@Property
+@FieldDefinition
+public class OnEntryAction implements BPMNProperty {
+
+    @Caption
+    public static final transient String caption = "On Entry Action";
+
+    @org.kie.workbench.common.stunner.core.definition.annotation.Description
```
Comment (pefernan) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/OnEntryAction.java]: I'd suggest to use an import isntead of using the full qualified name.

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/OnExitAction.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+@Portable
+@Bindable
+@Property
+@FieldDefinition
+public class OnExitAction implements BPMNProperty {
+
+    @Caption
+    public static final transient String caption = "On Exit Action";
+
+    @org.kie.workbench.common.stunner.core.definition.annotation.Description
```
Comment (pefernan) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/definition/property/task/OnExitAction.java]: same here, use import if possible.

**Comment by tsurdilo** [comment]
@pefernan amended - thanks!

**Comment by pefernan** [comment]
+1 for the forms side

**Comment by tsurdilo** [comment]
jenkins please retest this

**Comment by tsurdilo** [comment]
jenkins fail unrelated to this module. will ask to retest little later

**Comment by manstis** [comment]
See https://github.com/uberfire/uberfire/commit/16639e0a7dc66a77bedfa47c1083a429199bbbc4

**Comment by manstis** [comment]
Waiting for @ederign to revert UF commit and @porcelli to submit a replacement keeping the ```Observer``` mechanism.

**Comment by tsurdilo** [comment]
jenkins please retest this

**Comment by tsurdilo** [comment]
jenkins please retest this

**Comment by tsurdilo** [comment]
@manstis i think this is good to be merged now.

**Pull request merged**